File: app/src/main/java/pl/mareklangiewicz/myhub/ui/NotesAdapter.java
package pl.mareklangiewicz.myhub.ui;

import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import javax.inject.Inject;

import pl.mareklangiewicz.myhub.data.Note;
import pl.mareklangiewicz.myhub.R;


public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private @Nullable List<Note> notes;

    @Inject
    public NotesAdapter() { }

    public NotesAdapter(@Nullable List<Note> notes) {
        this.notes = notes;
    }

    public void setNotes(@Nullable List<Note> notes) {
        this.notes = notes;
        notifyDataSetChanged();
    }

    public @Nullable List<Note> getNotes() { return notes; }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.mg_note, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        //noinspection ConstantConditions
        Note note = notes.get(position);
        holder.mLabelTextView.setText(note.getLabel());
        holder.mTextTextView.setText(note.getText());
    }

    @Override
    public int getItemCount() {
        return notes == null ? 0 : notes.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public TextView mLabelTextView;
        public TextView mTextTextView;

        public ViewHolder(View itemView) {
            super(itemView);
            mLabelTextView = (TextView) itemView.findViewById(R.id.label);
            mTextTextView = (TextView) itemView.findViewById(R.id.text);
        }
    }
}

File: app/src/main/java/pl/mareklangiewicz/myhub/ui/NotesAdapter.kt
package pl.mareklangiewicz.myhub.ui

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.mh_item_note.view.*
import pl.mareklangiewicz.myhub.R
import pl.mareklangiewicz.myhub.data.Note
import pl.mareklangiewicz.myhub.mvp.INotesView

internal class NotesAdapter : RecyclerView.Adapter<NotesAdapter.ViewHolder>, INotesView {

    constructor() { }

    override var notes: List<Note> = emptyList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.mh_item_note, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val note = notes[position]
        holder.label.text = note.label ?: ""
        holder.text.text = note.text ?: ""
    }

    override fun getItemCount() = notes.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var label = itemView.mh_in_tv_label
        var text = itemView.mh_in_tv_text
    }
}
